api/src/test/java/io/druid/indexer/TaskStatusPlusTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexer;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+import io.druid.java.util.common.DateTimes;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class TaskStatusPlusTest
+{
+  @Test
+  public void testSerde() throws IOException
+  {
+    final ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(
+        new SimpleModule()
+            .addDeserializer(DateTime.class, new DateTimeDeserializer())
+            .addSerializer(DateTime.class, ToStringSerializer.instance)
+    );
+    final TaskStatusPlus status = new TaskStatusPlus(
+        "testId",
+        "testType",
+        DateTime.now(),
+        DateTime.now(),
+        TaskState.RUNNING,
+        1000L,
+        TaskLocation.create("testHost", 1010, -1)
+    );
+    final String json = mapper.writeValueAsString(status);
+    System.out.println(json);
+    Assert.assertEquals(status, mapper.readValue(json, TaskStatusPlus.class));
+  }
+
+  // Copied from io.druid.jackson.JodaStuff
```

Comment:
Instead of copying this, try making the objectMapper a `new DefaultObjectMapper()` (or use `TestHelper.makeJsonMapper`)

Reply:
Thanks, but it's not easy because of the cyclic dependency.

Reply:
Oh, sorry, I didn't notice this was in `druid-api`. Ok, carry on then.